DataFormats/MuonReco/src/MuonSelectors.cc

```diff
@@ -591,10 +593,11 @@ bool muon::isGoodMuon(const reco::Muon& muon,
 
         const float dX = std::abs(me0X - trkX);
         const float dY = std::abs(me0Y - trkY);
-        const float pullX = dX / std::sqrt(errX2 + me0ErrX2);
-        const float pullY = dY / std::sqrt(errY2 + me0ErrY2);
+        const float invPullX2 = errX2 + me0ErrX2;
+        const float invPullY2 = errY2 + me0ErrY2;
 
-        if ((dX < maxAbsDx or pullX < maxAbsPullX) and (dY < maxAbsDy or pullY < maxAbsPullY)) {
+        if ((dX < maxAbsDx or dX < maxAbsPullX * std::sqrt(invPullX2)) and
```

Comment:
dx*dx < maxAbsPullX2 *invPullX2
etc..

Reply:
@VinInn I would prefer to differ this to another possible PR in which errors and pulls in the muon DataFormat's are replaced with their squares, thus avoiding to compute square roots during the computation and having to square them again when used (quite often they are summed in quadrature in the calling code). Since that will imply some DataFormat change to be propagated around, I would prefer to factorize it and first check that this first part does not regress, and maybe also actually provides some (even little) improvement to the performance